macros/src/lib.rs

```diff
@@ -0,0 +1,98 @@
+// This file is part of radicle-link
+// <https://github.com/radicle-dev/radicle-link>
+//
+// Copyright (C) 2019-2020 The Radicle Team <dev@radicle.xyz>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 or
+// later as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#![feature(proc_macro_diagnostic)]
+
+use std::convert::TryFrom;
+
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, LitStr};
+
+use radicle_git_ext::reference::name::{RefLike, RefspecPattern};
+
+/// Create `RefLike` from a string literal.
+///
+/// The string is validated at compile time, and an unsafe conversion is
+/// emitted.
+///
+/// ```rust
+/// use std::convert::TryFrom;
```

Comment:
Not sure if there is a way to avoid having to import `TryFrom` at the use site -- it needs to be in scope at runtime. Maybe just provide `RefLike::unsafe_from(s: &str)`?